### ISSUE TITLE
[BUG] .tail() .head() fail in version 0.17.0 and above

### ISSUE THREAD

**Issue by stuart-wales-work** [unit8co/darts#817]
**Describe the bug**
Version: u8darts 0.17.0 and above

Timeseries subsetting methods .tail() and .head() fail, triggering error

  _File "C:\Users\stuartw\AppData\Roaming\Python\Python38\site-packages\darts\timeseries.py", line 1243, in head
    return self.__class__(self._xa[{axis_str: display_n}])
  File "C:\Users\stuartw\AppData\Roaming\Python\Python38\site-packages\darts\timeseries.py", line 142, in __init__
    raise_log(
  File "C:\Users\stuartw\AppData\Roaming\Python\Python38\site-packages\darts\logging.py", line 135, in raise_log
    raise exception_
_ValueError: The time dimension of the DataArray must be indexed either with a DatetimeIndex or with an RangeIndex_

this occurs no matter what length of subset is requested.
dataframe index is RangeIndex type and monotonic per .is_monotonic check in Pandas.

This error was not triggered in version 0.16.0

**To Reproduce**
simple example code

import pandas as pd
from darts import TimeSeries

df_full = pd.read_csv("1002.RNN.csv")

ts_pend=TimeSeries.from_dataframe(df_full).head()

Where 1002.RNN.csv is a csv with columns and rows of unheadered floats.

**Expected behavior**
create a subseries of the original timeseries.

**System (please complete the following information):**
Python 

3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)]

absl-py                 0.15.0
aiohttp                 3.8.1
aiosignal               1.2.0
async-timeout           4.0.2
attrs                   21.4.0
backcall                0.2.0
cachetools              4.2.4
certifi                 2021.10.8
charset-normalizer      2.0.7
click                   8.0.3
cmdstanpy               0.9.68
colorama                0.4.4
convertdate             2.3.2
cycler                  0.11.0
Cython                  0.29.24
decorator               5.1.0
Deprecated              1.2.13
ephem                   4.1
filelock                3.4.2
filterpy                1.4.5
frozenlist              1.3.0
fsspec                  2022.2.0
future                  0.18.2
google-auth             2.3.3
google-auth-oauthlib    0.4.6
grpcio                  1.41.1
hijri-converter         2.2.2
holidays                0.11.3.1
idna                    3.3
importlib-resources     5.4.0
ipython                 7.29.0
jedi                    0.18.0
joblib                  1.1.0
jsonschema              4.4.0
kiwisolver              1.3.2
korean-lunar-calendar   0.2.1
lightgbm                3.3.1
LunarCalendar           0.0.9
Markdown                3.3.4
matplotlib              3.4.3
matplotlib-inline       0.1.3
mpmath                  1.2.1
msgpack                 1.0.3
multidict               6.0.2
nfoursid                1.0.0
numpy                   1.21.4
oauthlib                3.1.1
OrcFxAPI                11.2.1
packaging               21.3
pandas                  1.3.4
parso                   0.8.2
patsy                   0.5.2
pickleshare             0.7.5
Pillow                  8.4.0
pip                     21.3.1
pmdarima                1.8.4
prompt-toolkit          3.0.22
protobuf                3.19.1
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pyDeprecate             0.3.1
Pygments                2.10.0
PyMeeus                 0.5.11
pyparsing               3.0.7
pyrsistent              0.18.1
pystan                  2.19.1.1
python-dateutil         2.8.1
pytorch-lightning       1.5.5
pytz                    2019.3
PyWavelets              1.1.1
PyYAML                  6.0
ray                     1.10.0
redis                   4.1.2
requests                2.26.0
requests-oauthlib       1.3.0
rsa                     4.7.2
scikit-learn            1.0.1
scipy                   1.7.1
setuptools              59.5.0
setuptools-git          1.2
six                     1.13.0
statsmodels             0.13.0
sympy                   1.8
tensorboard             2.7.0
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.0
threadpoolctl           3.0.0
torch                   1.10.0
torchmetrics            0.7.2
tqdm                    4.62.3
traitlets               5.1.1
typing-extensions       3.10.0.2
u8darts                 0.17.0
ujson                   4.2.0
urllib3                 1.26.7
wcwidth                 0.2.5
Werkzeug                2.0.2
wheel                   0.37.0
wrapt                   1.13.3
xarray                  0.20.1
yarl                    1.7.2
zipp                    3.7.0

**Additional context**
Depreciating u8darts to 0.16.0 results in incompatabilities using previously trained models (trained in u8darts 0.17.1)

**Comment by hrzn** [comment]
Thank you for reporting, seems like a bug indeed.

**Comment by stuart-wales-work** [comment]
No, thank you for your awesome library!

Best Regards

Stuart Wales
Director
Head: Marine and Structures Section
Email Reference  |
AMOG Pty Ltd
19 Business Park Drive, Notting Hill Vic 3168, Australia
T +61 3 9542 3700   |  M  +61 (0) 432 872787
E ***@***.*** ***@***.***>
W www.amog.consulting <http://www.amog.consulting/>
Australia  |  United States  |  Colombia  |  United Kingdom  |  Malaysia
***@***.***
This e-mail is intended only for the addressee or addressees named above and associated project team members, as dictated by project requirements, and may contain confidential information. If you are not the intended recipient, please let us know immediately by return e-mail and then delete this e-mail, without disclosing or copying the contents. The content of emails to and from AMOG Consulting is monitored for corporate governance purposes. Emails addressed to individuals within AMOG may be copied to other AMOG staff as dictated by project requirements. AMOG does not accept liability for any errors or omissions in the content of this message which may arise as a result of the email transmission. We recommend that you confirm correct transmission by reference to the original in hard copy form or other means before relying on information contained herein for design, construction or operational purposes

**Comment by stuart-wales-work** [comment]
Thanks Julien, looking forward to checking it out.

Best Regards

Stuart Wales
